File: app/src/main/java/com/loopeer/codereader/sync/DownloadReceiver.java
package com.loopeer.codereader.sync;

import android.app.DownloadManager;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.loopeer.codereader.Navigator;

public class DownloadReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        if (intent.hasExtra(DownloadManager.EXTRA_DOWNLOAD_ID)) {
            long downloadId = intent.getLongExtra(
                    DownloadManager.EXTRA_DOWNLOAD_ID, 0);
            if (downloadId > 0) {
                Intent i = new Intent(context, DownloadRepoService.class);
                i.putExtra(DownloadManager.EXTRA_DOWNLOAD_ID, downloadId);
                context.startService(i);
            }
        }

        if (intent.hasExtra(DownloadManager.EXTRA_NOTIFICATION_CLICK_DOWNLOAD_IDS)) {
            Navigator.startMainActivity(context);
        }
    }

}

File: code-reader-kt/src/main/java/com/loopeer/codereaderkt/sync/DownloadReceiver.kt
package com.loopeer.codereaderkt.sync

import android.app.DownloadManager
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.loopeer.codereaderkt.Navigator


class DownloadReceiver : BroadcastReceiver() {
    override fun onReceive(p0: Context?, p1: Intent?) {
        if (p1!!.hasExtra(DownloadManager.EXTRA_DOWNLOAD_ID)) {
            var download: Long = p1.getLongExtra(
                    DownloadManager.EXTRA_DOWNLOAD_ID, 0
            )
            if (download > 0) {
                var i: Intent = Intent(p0, DownloadRepoService::class.java)
                i.putExtra(DownloadManager.EXTRA_DOWNLOAD_ID, download)
                i.putExtra(Navigator.EXTRA_DOWNLOAD_SERVICE_TYPE, DownloadRepoService.DOWNLOAD_COMPLETE)
                p0!!.startService(i)
            }
        }
    }

}